nginx-controller/controller/controller.go

```diff
@@ -681,12 +681,37 @@ func (lbc *LoadBalancerController) syncIng(task Task) {
 		glog.V(2).Infof("Adding or Updating Ingress: %v\n", key)
 
 		ing := obj.(*extensions.Ingress)
+
+		if mergeableType, exists := ing.Annotations["nginx.org/mergible-ingress-type"]; exists {
+			if mergeableType == "master" {
+				mergableIngExs, err := lbc.validateMergableType(*ing)
+				if err != nil {
+					lbc.recorder.Eventf(ing, api_v1.EventTypeWarning, "AddedOrUpdatedWithError", "Configuration for %v was added or updated, but not applied: %v", key, err)
+					return
+				}
+				err = lbc.cnf.AddOrUpdateMergableIngress(mergableIngExs)
+				if err != nil {
+					lbc.recorder.Eventf(ing, api_v1.EventTypeWarning, "AddedOrUpdatedWithError", "Configuration for %v was added or updated, but not applied: %v", key, err)
+				} else {
+					lbc.recorder.Eventf(ing, api_v1.EventTypeNormal, "AddedOrUpdated", "Configuration for %v was added or updated", key)
+				}
+				return
+			} else if mergeableType == "minion" {
+				// We ignore any minions, they are processed whenever a master is
```

Comment:
We can't simply ignore it. Consider the case when a minion is added. This will be ignored.

The correct solution for handling minions is to find their masters and process the master. If there is no master, only than we should Ignore it.

Reply:
Makes Sense.